graylog2-web-interface/src/views/pages/NewDashboardPage.jsx

```diff
@@ -1,63 +1,44 @@
 // @flow strict
-import React, { useEffect, useState } from 'react';
+import * as React from 'react';
+import { useMemo } from 'react';
 import PropTypes from 'prop-types';
 
 import Spinner from 'components/common/Spinner';
-import type { ViewHook } from 'views/logic/hooks/ViewHook';
-import { processHooks } from 'views/logic/views/ViewLoader';
 import withPluginEntities from 'views/logic/withPluginEntities';
 import viewTransformer from 'views/logic/views/ViewTransformer';
 import { ViewActions } from 'views/stores/ViewStore';
 import View from 'views/logic/views/View';
-import type { ViewJson } from 'views/logic/views/View';
 import { ExtendedSearchPage } from 'views/pages';
 import { IfPermitted } from 'components/common';
+import useLoadView from 'views/logic/views/UseLoadView';
 
 type Props = {
   route: {},
   location: {
     state?: {
-      view?: View | ViewJson,
+      view?: View,
     },
     query: { [string]: any },
-  };
-  loadingViewHooks: Array<ViewHook>,
-  executingViewHooks: Array<ViewHook>,
+  },
 };
-const NewDashboardPage = ({ route, location, loadingViewHooks, executingViewHooks }: Props) => {
-  const [loaded, setLoaded] = useState(false);
-  const [hookComponent, setHookComponent] = useState(undefined);
 
-  useEffect(() => {
-    let mounted = true;
-    const { state = {} } = location;
-    const { view: searchView } = state;
-
-    if (searchView && searchView.search) {
-      const { query } = location;
-      /* $FlowFixMe the searchView.search is guard enough and instanceof does not work here */
+const NewDashboardPage = ({ route, location }: Props) => {
```

Comment:
For the `ShowViewPage`, which is being used for saved dashboards, we are using the `SavedSearchPage`.

I am wondering if we should implement the `SavedSearchPage` for this component as well.
We do not need the `ViewLoaderContext` and `NewViewLoaderContext`, but we would use `IfUserHasAccessToAnyStream` here as well. Or am I missing a reason why we should not use `IfUserHasAccessToAnyStream` here?

Reply:
:white_check_mark: